src/sentry/utils/types.py

```diff
@@ -181,7 +181,41 @@ def convert(self, value):
 }
 
 
-def type_from_value(value: typing.Any) -> typing.Callable[[typing.Any], typing.Any]:
+# @typing.overload
+# def type_from_value(value: bool) -> BoolType:
```

Comment:
what's wrong with bool?

Reply:
If this was left uncommented mypy would complain about overloads conflicting. I don't understand why though as int and bool are separate types.

Reply:
Related to python's [weird bool inheritance](https://peps.python.org/pep-0285/) maybe?

```python
>>> isinstance(True, int)
True
```

Reply:
This works well:
```python
import typing


class A: pass
class B: pass

@typing.overload
# type: ignore[misc]  # bool overlaps with int
def f(x: bool) -> B: ...
@typing.overload
def f(x: int) -> A: ...

def f(x):
    if isinstance(x, bool): return A()
    elif isinstance(x, str): return B()
    else: raise TypeError(type(x))

reveal_type(f(1))   # Revealed type is "demo.A"
reveal_type(f(True))  # Revealed type is "demo.B"
```

As far as I can see, this error is plain old wrong, and things start working correctly with it disabled.